tests/test_config.py

```diff
@@ -73,3 +90,165 @@ def test_reference_anchor_for_module(repo):
+
+    assert file_config.config["people"] == config["people"]
+
+    file_config.config["people"].pop()
+    file_config.write()
+
+    clean(file_config, refresh=True)
+
+    cleaned_and_sorted = sorted(file_config.config["people"], key=itemgetter("bmo_id"))
+    assert len(cleaned_and_sorted) == 3
+    for i, person in enumerate(cleaned_and_sorted):
+        assert person["name"] == test_bmo_user_data[i]["real_name"]
+        assert person["nick"] == test_bmo_user_data[i]["nick"]
+
+
+@mock.patch("mots.config.get_bmo_data")
+def test_clean_changed_names(get_bmo_data, repo, config, test_bmo_user_data):
+    """Test that updated BMO data will be reflected when cleaning."""
+    get_bmo_data.return_value = test_bmo_user_data
+
+    file_config = FileConfig(repo / "mots.yml")
+    file_config.load()
+
+    assert file_config.config["people"] == config["people"]
+
+    clean(file_config)
+
+    for person in file_config.config["people"]:
+        assert person["name"] == test_bmo_user_data[person["bmo_id"]]["real_name"]
+        assert person["nick"] == test_bmo_user_data[person["bmo_id"]]["nick"]
+
+
+@mock.patch("mots.config.get_bmo_data")
+def test_clean_changed_names_no_refresh(get_bmo_data, repo, config, test_bmo_user_data):
+    """Test that updated BMO data has no effect when cleaning without refresh."""
+    get_bmo_data.return_value = test_bmo_user_data
+
+    file_config = FileConfig(repo / "mots.yml")
+    file_config.load()
+
+    assert file_config.config["people"] == config["people"]
+
+    clean(file_config, refresh=False)
+
+    assert sorted(file_config.config["people"], key=itemgetter("bmo_id")) == sorted(
+        config["people"], key=itemgetter("bmo_id")
+    )
+
+
+@mock.patch("mots.config.get_bmo_data")
+def test_clean_added_user_no_refresh(get_bmo_data, repo, config, test_bmo_user_data):
+    """Test that updated BMO data has no effect when cleaning without refresh."""
+    get_bmo_data.return_value = test_bmo_user_data
+
+    file_config = FileConfig(repo / "mots.yml")
+    file_config.load()
+
+    assert file_config.config["people"] == config["people"]
+
+    file_config.config["people"].insert(1, {"bmo_id": 3})
+    file_config.write()
+
+    assert len(file_config.config["people"]) == 5
+
+    # Check that the new entry is unaffected after write.
+    for person in file_config.config["people"]:
+        if person["bmo_id"] == 3:
+            # Ensure only the bmo_id key is there.
+            assert len(person) == 1
+
+    # Compare the old and new people lists without the new entry, they should match.
+    new = sorted(
+        [dict(p) for p in file_config.config["people"] if p["bmo_id"] != 3],
+        key=itemgetter("bmo_id"),
+    )
+    original = sorted([p for p in config["people"]], key=itemgetter("bmo_id"))
+    assert new == original
+
+    # Run clean without refresh.
+    clean(file_config, refresh=False)
+
+    # It is expected that only the new entry is updated, and the rest are intact.
```

Comment:
This comment would be better as part of the test, like:

```python
assert cleaned[0] == original[0], "Unchanged entry should not be updated after clean"
assert cleaned[3] == {"bmo_id": 3, "name": "seven", "nick": "nanotubes"}, "New entry should be updated from BMO."
```

etc. Then they appear directly in pytest output if the test fails.